hwilib/hwwclient.py

```diff
@@ -1,82 +1,167 @@
-    def get_pubkey_at_path(self, path):
-        raise NotImplementedError('The HardwareWalletClient base class does not '
-                                  'implement this method')
-
-    # Must return a hex string with the signed transaction
-    # The tx must be in the combined unsigned transaction format
-    def sign_tx(self, tx):
-        raise NotImplementedError('The HardwareWalletClient base class does not '
-                                  'implement this method')
-
-    # Must return a base64 encoded string with the signed message
-    # The message can be any string. keypath is the bip 32 derivation path for the key to sign with
-    def sign_message(self, message, keypath):
-        raise NotImplementedError('The HardwareWalletClient base class does not '
-                                  'implement this method')
-
-    # Display address of specified type on the device.
-    def display_address(self, keypath, p2sh_p2wpkh, bech32, redeem_script=None):
-        raise NotImplementedError('The HardwareWalletClient base class does not '
-                                  'implement this method')
-
-    # Setup a new device
-    def setup_device(self, label='', passphrase=''):
-        raise NotImplementedError('The HardwareWalletClient base class does not '
-                                  'implement this method')
-
-    # Wipe this device
-    def wipe_device(self):
-        raise NotImplementedError('The HardwareWalletClient base class does not '
-                                  'implement this method')
-
-    # Restore device from mnemonic or xprv
-    def restore_device(self, label='', word_count=24):
-        raise NotImplementedError('The HardwareWalletClient base class does not implement this method')
-
-    # Begin backup process
-    def backup_device(self, label='', passphrase=''):
-        raise NotImplementedError('The HardwareWalletClient base class does not implement this method')
-
-    # Close the device
-    def close(self):
-        raise NotImplementedError('The HardwareWalletClient base class does not '
-                                  'implement this method')
-
-    # Prompt pin
-    def prompt_pin(self):
-        raise NotImplementedError('The HardwareWalletClient base class does not implement this method')
-
-    # Send pin
-    def send_pin(self):
-        raise NotImplementedError('The HardwareWalletClient base class does not implement this method')
-
-    # Toggle passphrase
-    def toggle_passphrase(self):
-        raise NotImplementedError('The HardwareWalletClient base class does not implement this method')
+    def get_pubkey_at_path(self, bip32_path: str) -> Dict[str, str]:
+        """Return a dictionary with the xpub.
+
+        Retrieve the public key at the specified BIP32 derivation path.
+
+        Return {"xpub": xpub}
+        """
+        raise NotImplementedError("not implemented by HardwareWalletClient base class")
+
+    def sign_tx(self, tx) -> str:
```

Comment:
This actually returns a `Dict[str, str]`. All of these functions should return a dict.